Enable TensorPipe CUDA fallback channel

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50676 Let TensorPipe detect peer access
* **#50675 Enable TensorPipe CUDA fallback channel**
* #50674 Enable TensorPipe CUDA sending to self

Differential Revision: [D25941963](https://our.internmc.facebook.com/intern/diff/D25941963)

mrshenli committed:
Enable TensorPipe CUDA fallback channel

[ghstack-poisoned]

lw commented on torch/csrc/distributed/rpc/tensorpipe_agent.cpp:
```diff
@@ -86,6 +86,8 @@ constexpr int64_t kBasicChannelPriority = 0;
 #if TENSORPIPE_HAS_CUDA_IPC_CHANNEL && defined(USE_CUDA_NOT_ROCM)
 constexpr int64_t kCudaIpcChannelPriority = 300;
 constexpr int64_t kCudaXthChannelPriority = 400;
+constexpr int64_t kCudaBasicChannelPriority = 500;
```
Same here, shouldn't be guarded by `TENSORPIPE_HAS_CUDA_IPC_CHANNEL`.

lw replied:
Oh and actually this should have a much lower priority, like 100. Otherwise we'll always prefer sending over TCP rather than doing IPC.

The final priority order should be: BASIC < GDR < IPC < XTH.

mrshenli committed:
Update on "Enable TensorPipe CUDA fallback channel"


Differential Revision: [D25941963](https://our.internmc.facebook.com/intern/diff/D25941963)

[ghstack-poisoned]

mrshenli committed:
Update on "Enable TensorPipe CUDA fallback channel"


Differential Revision: [D25941963](https://our.internmc.facebook.com/intern/diff/D25941963)

[ghstack-poisoned]
